### PR TITLE
[#10199] fix No message when sending funds in 1-1 chat with keycard

### LINKED ISSUE #10199: No message when sending funds  in 1-1 chat with keycard
# Bug Report

## Problem
Message is not shown in 1-1 chat after sending transaction.

#### Expected behavior
Message is not shown in 1-1 chat after sending transaction in 1-1 chat

#### Actual behavior
No message after signing transaction

### Reproduction

[comment]: # (Describe how we can replicate the bug step by step.)

- Open Status
- Start 1-1 chat with ENS
- Try to send some ETH (i.e. 0.01)

### Additional Information

- Status version: nightly 23/03/2020
- Operating System: Android

#### Logs

[Status-debug-logs-10.zip](https://github.com/status-im/status-react/files/4369994/Status-debug-logs-10.zip)
Video: https://drive.google.com/file/d/1IH2y17NdSsxN-h3iwkHASy62ZGBPJ1rx/view?usp=sharing

### PR DESCRIPTION
**Pull request by rasom**
fix #10199 

status: ready

### CODE REVIEW

File: src/status_im/signing/core.cljs [hunk 1]
```diff
@@ -239,25 +244,67 @@
               #(when on-result
                  {:dispatch (conj on-result result)}))))
 
+(fx/defn sign-message-foo
```
Comment (flexsurfer) [inline src/status_im/signing/core.cljs]: foo?

Comment (rasom) [reply src/status_im/signing/core.cljs]: right, leftover, thanks :D

Comment (rasom) [reply src/status_im/signing/core.cljs]: fixed

File: src/status_im/signing/core.cljs [hunk 2]
```diff
@@ -239,25 +244,67 @@
               #(when on-result
                  {:dispatch (conj on-result result)}))))
 
+(fx/defn sign-hashed-message
+  {:events [::sign-message]}
```
Comment (flexsurfer) [inline src/status_im/signing/core.cljs]: this should be in hardwallet ns then?

File: src/status_im/signing/core.cljs [hunk 3]
```diff
@@ -239,25 +244,67 @@
...
+  [cofx params result]
+  (let [{:keys [result error]} (types/json->clj result)]
+    (hardwallet.sign/sign-message
+     cofx
+     {:hash       (ethereum/naked-address result)
+      :on-success #(re-frame/dispatch [::on-sign-message-success params %])})))
+
+(fx/defn on-sign-message-success
+  {:events [::on-sign-message-success]}
```
Comment (flexsurfer) [inline src/status_im/signing/core.cljs]: also should be in hardwallet ns?

**Review by flexsurfer — APPROVED** [review]

File: src/status_im/signing/core.cljs (hunk 3 above)
Comment (rasom) [reply src/status_im/signing/core.cljs]: fixed

File: src/status_im/signing/core.cljs (hunk 2 above)
Comment (rasom) [reply src/status_im/signing/core.cljs]: fixed

**Comment by churik** [comment]
@rasom 
tested on build https://github.com/status-im/status-react/commit/9ef06f4400af9c492af7a077b58152142a6de82d - after "Sign and send" with keycard state of message isn't changed to 'Pending"

Monosnap 2020-04-06 11-53-10

**Steps:**
- create account with keycard
- as sender: create 1-1 with receiver
- as sender: try to send some ETH
- as receiver: Accept and share addredd
- as sender: Sign and send transaction

Network: Mainnet

**Expected result:**
transaction is sent, state in 1-1 is "Pending"

**Actual result:**
transaction is sent, state in 1-1 is not changed

**Comment by rasom** [comment]
@churik is that screen on sender or receiver side?

**Comment by churik** [comment]
@rasom on Sender side, sorry for confusion

**Comment by churik** [comment]
@rasom 
https://github.com/status-im/status-react/pull/10264#issuecomment-609695671 is still reproducible with ordinary accounts, but it is not with keycard accounts.

Full log and video for sender side is [here](https://app.saucelabs.com/tests/60ad9374edab4812a0064d71dea853f8?auth=8b34f041703ff8e35a5e6c0d87002f0f#118).

**Comment by churik** [comment]
"Confirmed" state is not shown in case if  ETH was sent from keycard account (for ordinary accounts all good)
Will be logged separately.

**Pull request merged**